bundles/org.openhab.binding.homeconnect/src/main/java/org/openhab/binding/homeconnect/internal/client/HomeConnectApiClient.java

```diff
@@ -0,0 +1,1134 @@
+        communicationQueue = QueueUtils.synchronizedQueue(new CircularFifoQueue<>(COMMUNICATION_QUEUE_SIZE));
+        if (apiRequestHistory != null) {
+            communicationQueue.addAll(apiRequestHistory);
+        }
+    }
+
+    /**
+     * Get all home appliances
+     *
+     * @return list of {@link HomeAppliance}
+     * @throws CommunicationException API communication exception
+     * @throws AuthorizationException oAuth authorization exception
+     */
+    public List<HomeAppliance> getHomeAppliances() throws CommunicationException, AuthorizationException {
+        Request request = createGetRequest("/api/homeappliances");
+        try (Response response = client.newCall(request).execute()) {
+            checkResponseCode(HTTP_OK, request, response, null, null);
+
+            String responseBody = mapToString(response.body());
+            trackAndLogApiRequest(null, request, null, response, responseBody);
+
+            return mapToHomeAppliances(responseBody);
+        } catch (IOException | ApplianceOfflineException e) {
+            logger.warn("Failed to fetch home appliances! error={}", e.getMessage());
+            trackAndLogApiRequest(null, request, null, null, null);
+            throw new CommunicationException(e);
+        }
+    }
+
+    /**
+     * Get home appliance by id
+     *
+     * @param haId home appliance id
+     * @return {@link HomeAppliance}
+     * @throws CommunicationException API communication exception
+     * @throws AuthorizationException oAuth authorization exception
+     */
+    public HomeAppliance getHomeAppliance(String haId) throws CommunicationException, AuthorizationException {
+        Request request = createGetRequest("/api/homeappliances/" + haId);
+        try (Response response = client.newCall(request).execute()) {
+            checkResponseCode(HTTP_OK, request, response, haId, null);
+
+            String responseBody = mapToString(response.body());
+            trackAndLogApiRequest(haId, request, null, response, responseBody);
+
+            return mapToHomeAppliance(responseBody);
+        } catch (IOException | ApplianceOfflineException e) {
+            logger.warn("Failed to get home appliance! haId={}, error={}", haId, e.getMessage());
+            trackAndLogApiRequest(haId, request, null, null, null);
+            throw new CommunicationException(e);
+        }
+    }
+
+    /**
+     * Get ambient light state of device.
+     *
+     * @param haId home appliance id
+     * @return {@link Data}
+     * @throws CommunicationException API communication exception
+     * @throws AuthorizationException oAuth authorization exception
+     * @throws ApplianceOfflineException appliance is not connected to the cloud
+     */
+    public Data getAmbientLightState(String haId)
+            throws CommunicationException, AuthorizationException, ApplianceOfflineException {
+        return getSetting(haId, "BSH.Common.Setting.AmbientLightEnabled");
```

Comment:
Why not using the constant you defined in HomeConnectBindingConstants ?
Same comment for other methods.

Reply:
I used the API client class in another private project. Will adjust it. It's better to use constants.